### ISSUE TITLE
can't user local HTTP proxy

### ISSUE THREAD

**Issue by wenqiangxie** [UnigramDev/Unigram#1224]
The last version form MS Store can't use local HTTP proxy(from 127.0.0.1 or localhost), but before version can use that. I have test the localhost proxy server by telegram desktop version(from MS Store), every things show that proxy server works normal. Could you fix? It's is a really important feature in some where.

**Comment by Tirraon** [comment]
> Could you fix? 

I think, you can.
Because I am almost sure, that problem is same as here: https://stackoverflow.com/questions/33259763/uwp-enable-local-network-loopback

**Comment by wenqiangxie** [comment]
> > Could you fix?
> 
> I think, you can.
> Because I am almost sure, that problem is same as here: https://stackoverflow.com/questions/33259763/uwp-enable-local-network-loopback

Thank you. I will test that solution. But I remember  before version can user local proxy without any more setting, so I want to sure may be Unigram have some way to use local proxy or all UWP applications can't do that and some mistakes in my memory. Thank you again.

**Comment by FrayxRulez** [comment]
No, it has never been possible to use local proxy without enabling the local loopback. Sure thing Unigram might inform the users about this, when they try to.

**Comment by thawisakprasreod** [comment]
@Tirraon

**Comment by BuiltByRaj** [comment]
any fix?

**Comment by FrayxRulez** [comment]
> any fix?

Yes, check users suggestions

**Comment by william0620** [comment]
So any enhancement?

**Comment by Weirdo1312** [comment]
i still have this issue

**Comment by makisukurisu** [comment]
Hi, @Weirdo1312, have you tried enabling local loop back?

If not – try the following command:
```
Get-AppxPackage | %{$_.PackageFamilyName} | where{$_ -match "unigram"} | %{CheckNetIsolation LoopbackExempt -a -n="$_"}
```
It was suggested long-long ago in one of the [messages in Unigram Insiders](https://t.me/unigraminsiders/480397)

If the problem persists – get back to us.

**Comment by Weirdo1312** [comment]
> Hi, @Weirdo1312, have you tried enabling local loop back?
> 
> If not – try the following command:
> 
> ```
> Get-AppxPackage | %{$_.PackageFamilyName} | where{$_ -match "unigram"} | %{CheckNetIsolation LoopbackExempt -a -n="$_"}
> ```
> 
> It was suggested long-long ago in one of the [messages in Unigram Insiders](https://t.me/unigraminsiders/480397)
> 
> If the problem persists – get back to us.

no, enabling it fixed it, thanks

**Comment by kamednokg-lang** [comment]
- 

- [ ]

**Comment by FrayxRulez** [comment]
The app now automatically executes `CheckNetIsolation` on every launch and hooks system proxy settings by default.

**Comment by Yosh7937** [comment]
UAndMi-AtWorkCo-Operation https://github.com/JetBrains/kotlin/pull/5613#discussion_r2649095554> The last version form MS Store can't use local HTTP proxy(from 127.0.0.1 or localhost), but before version can use that. I have test the localhost proxy server by telegram desktop version(from MS Store), every things show that proxy server works normal. Could you fix? It's is a really important feature in some where.

**Comment by RAFI7708** [comment]
Fixed that

**Comment by Neurotoxin001** [comment]
unigram
<img width="264" height="131" alt="Image" src="https://github.com/user-attachments/assets/3b609745-131d-4e91-8e53-04fa63febf9b" />

tdesktop
<img width="386" height="119" alt="Image" src="https://github.com/user-attachments/assets/bdc8eed4-b52d-4b6d-a394-b353e5077a96" />

why

**Comment by FrayxRulez** [comment]
@Neurotoxin001 pretty much explained in this ticket
